AirCastingTests/BluetoothProtectorTests.swift

```diff
@@ -0,0 +1,199 @@
+import XCTest
+import Resolver
+import Foundation
+@testable import AirCasting
+
+class BluetoothProtectorTests: ACTestCase {
+    
+    func test_whenAskedToPredicateStatuses_excludesUnsupported() throws {
+        let database = AirBeamDatabaseSpy()
+        Resolver.test.register { database as SessionsFetchable }
+        let sut = BluetoothConnectionProtector()
+        
+        sut.isAirBeamAvailableForNewConnection(completion: { _ in })
+        
+        XCTAssertEqual(database.constrainedCalls.count, 1)
+        let constraint = try XCTUnwrap(database.constrainedCalls.first)
+        guard case let .predicate(predicate) = constraint else { XCTFail(); return }
+        
+        let correctStatuses: [FakePredicateBlueprint] = [.createExpected(status: 0),
+                                                  .createExpected(status: 2),
+                                                  .createExpected(status: -1)]
+        // Status: 2 should be excluded
+        XCTAssertEqual(2, (correctStatuses as NSArray).filtered(using: predicate).count)
+        
+        let invalidStatuses: [FakePredicateBlueprint] = [.createExpected(status: -2),
+                                                         .createExpected(status: 3),
+                                                         .createExpected(status: 5)]
+        
+        XCTAssertEqual(0, (invalidStatuses as NSArray).filtered(using: predicate).count)
+    }
+    
+    func test_whenAskedToPredicateDeviceTypes_excludesUnsupported() throws {
+        let database = AirBeamDatabaseSpy()
+        Resolver.test.register { database as SessionsFetchable }
```

Comment:
I think those two lines could be moved to setup func

Reply:
I think we use Spy/Stub classes to double for `SessionsFetchable`, so it wouldn't work because we're switching between the two. What we can do is to move those Stub/Spy to properties.